src/compiler/scala/tools/nsc/ast/parser/Parsers.scala

```diff
@@ -95,6 +95,23 @@ trait ParsersCommon extends ScannersCommon { self =>
      */
     @inline final def makeParens(body: => List[Tree]): Parens =
       Parens(inParens(if (in.token == RPAREN) Nil else body))
+
+    /** {{{ part { `sep` part } }}}, or if sepFirst is true, {{{ { `sep` part } }}}. */
+    final def tokenSeparated[T](separator: Token, sepFirst: Boolean, part: => T): List[T] = {
```

Comment:
The trailing comma fix touches `tokenSeparated`; I can't tell offhand if this change works; the `ParserCommon`/`ScannerCommon` is impoverished.

Reply:
I meant to say whether this change breaks my change yesterday.

Reply:
Breaks how?  It's just moved so it can be shared.

Reply:
My change requires using `in.sepRegions` and `in.skipTrailingComma`, and changing to `ParserCommon` changes `in` from `Scanner` to `ScannerCommon`. I haven't tried merging yet, I'm just speculating, but my broader point is that the abstraction is more anemic.

Reply:
I guess I can always override tokenSeparated, etc. I added `skipTrailingComma` (with trivial impl) at the common level to avoid adding trivial impls for the other subclasses esp the JavaScanner.

Reply:
Actually `tokenSeparated` is also over-abstracted. Well, I guess there are more than two cases, there are three, including `AT`.

Reply:
(@som-snytt I'm resolving this, because I don't seem what I need to change.  But let me know if I'm missing something.)